core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1193,15 +1193,22 @@ private boolean checkConsistencyInternal(LockedInodePath inodePath) throws Inval
       if (ufs == null) {
         return true;
       }
+      UfsStatus ufsStatus;
+      try {
+        ufsStatus = ufs.getStatus(ufsPath);
+      } catch (FileNotFoundException e) {
+        return false;
```

Comment:
this should be `return !inode.isPersisted()` if its not in the ufs. you could return early after line 1189, with if (!inode.isPersisted) return true